p2p/src/channels/p2p_channels_effectful_effects.rs

```diff
@@ -0,0 +1,133 @@
+use openmina_core::bug_condition;
+use redux::ActionMeta;
+
+use crate::webrtc::{Offer, P2pConnectionResponse};
+
+use super::{
+    signaling::{
+        discovery::{P2pChannelsSignalingDiscoveryAction, SignalingDiscoveryChannelMsg},
+        exchange::{P2pChannelsSignalingExchangeAction, SignalingExchangeChannelMsg},
+    },
+    ChannelMsg, MsgId, P2pChannelsEffectfulAction, P2pChannelsService,
+};
+
+impl P2pChannelsEffectfulAction {
+    pub fn effects<Store, S>(self, _meta: &ActionMeta, store: &mut Store)
+    where
+        Store: crate::P2pStore<S>,
+        Store::Service: P2pChannelsService,
+    {
+        match self {
+            P2pChannelsEffectfulAction::InitChannel {
+                peer_id,
+                id,
+                on_success,
+            } => {
+                store.service().channel_open(peer_id, id);
+                store.dispatch_callback(on_success, peer_id);
+            }
+            P2pChannelsEffectfulAction::RequestSend {
```

Comment:
based on what @dkuehr mentioned in other comment, this probably needs a better name, otherwise it is confusing (as it is now we use `Request` to send responses too)